File: python/q1_plot.py
# python/q1_plot.py
from __future__ import annotations

from pathlib import Path
import pandas as pd
import matplotlib.pyplot as plt


def save_top5_barh(top5: pd.DataFrame, out_path: Path, title: str = "Top 5 by |corr| (first differences)") -> None:
    plot_df = top5.sort_values("abs_corr", ascending=True)

    plt.figure(figsize=(10, 4))
    plt.barh(plot_df["code"], plot_df["abs_corr"])
    plt.title(title)
    plt.xlabel("|Pearson correlation|")
    plt.ylabel("Candidate series code")
    plt.grid(True, axis="x")
    plt.tight_layout()

    out_path.parent.mkdir(parents=True, exist_ok=True)
    plt.savefig(out_path, dpi=200)
    plt.close()

File: R/q1_plot.R
# R/q1_plot.R

suppressWarnings({
  library(ggplot2)
})

plot_top_corr_barh <- function(top_df, title = "Top 5 by |corr| (first differences)") {
  # Expect columns: code, abs_corr
  ggplot(top_df, aes(x = abs_corr, y = reorder(code, abs_corr))) +
    geom_col() +
    labs(
      title = title,
      x = "|Pearson correlation|",
      y = "Candidate series code"
    ) +
    theme_minimal()
}

save_plot_png <- function(p, path, width = 10, height = 4, dpi = 200) {
  dir.create(dirname(path), recursive = TRUE, showWarnings = FALSE)
  ggsave(filename = path, plot = p, width = width, height = height, dpi = dpi)
}
